src/main/java/graphql/execution/instrumentation/Instrumentation.java

```diff
@@ -284,11 +300,27 @@ default InstrumentationContext<ExecutionResult> beginField(InstrumentationFieldP
      *
      * @return a nullable {@link InstrumentationContext} object that will be called back when the step ends (assuming it's not null)
      */
+    @Deprecated
+    @DeprecatedAt("2023-09-11")
     @Nullable
     default InstrumentationContext<ExecutionResult> beginField(InstrumentationFieldParameters parameters, InstrumentationState state) {
         return beginField(parameters.withNewState(state));
     }
 
+    /**
+     * This is called just before a field is resolved into a value.
+     *
+     * @param parameters the parameters to this step
+     * @param state      the state created during the call to {@link #createState(InstrumentationCreateStateParameters)}
+     *
+     * @return a nullable {@link InstrumentationContext} object that will be called back when the step ends (assuming it's not null)
+     */
+    @Nullable
+    default InstrumentationContext<Object> beginFieldExecution(InstrumentationFieldParameters parameters, InstrumentationState state) {
+        InstrumentationContext<ExecutionResult> ic = beginField(parameters, state);
+        return ic == null ? null : new ExecutionResultInstrumentationContextAdapter(ic);
+    }
+
```

Comment:
This is not a breaking change because the default implementation calls back to now deprecated `beginField` via the adapter

Reply:
I have updated the graphql-java shipped ones so that they dont do this old behavior